index.tsx

```diff
@@ -982,6 +1017,11 @@ export function compiler(
       )
     )
 
+    arr = arr.filter(item => item)
+    while (typeof(arr[arr.length - 1]) == "string" && !arr[arr.length - 1].trim()) {
+      arr = arr.slice(0, -1);
```

Comment:
In-place mutation is much faster.

```suggestion
      arr.pop();
```

Reply:
330f0440d4aa981e3c6652245ca0b078811b48e1